pages/product/[name].js

```diff
@@ -0,0 +1,117 @@
+import Link from 'next/link'
+import NextImage from 'next/image'
+import { useState, useEffect } from 'react'
+import { StarIcon } from '@heroicons/react/solid'
+import { relativizeURL, shimmer, toBase64 } from '@/components/Shimmer'
+import { HeartIcon, StarIcon as StarIconOutline } from '@heroicons/react/outline'
+
+const Product = ({ data }) => {
+  const [relatedProducts, setRelatedProducts] = useState([])
+  const [selectedImage, setSelectedImage] = useState(data.images[0].url)
```

Comment:
If you use an index, this can just be `0`

`const [selectedImage, setSelectedImage] = useState(0)`

Reply:
Adding that!